packages/plugin-ext/src/plugin/plugin-manager.ts

```diff
@@ -219,6 +221,12 @@ export class PluginManagerExtImpl implements PluginManagerExt, PluginManager {
 
         this.webview.init(params.webview);
         this.jsonValidation = params.jsonValidation;
+
+        this.supportedActivationEvents = new Set(PluginManagerExtImpl.BUILTIN_ACTIVATION_EVENTS);
+        const additionalActivationEvents = await this.envExt.getEnvVariable(PluginManagerExtImpl.ADDITIONAL_ACTIVATION_EVENTS_ENV);
```

Comment:
I wonder whether the additional activation events shouldn't be part of the `PluginManagerInitializeParams` used in the $init call to initialize the plugin manager. It already contains the some other setup data and is also called before the $start method with the `PluginManagerStartParams` where we get the first set of activation events. I think that would also give us a more fine-grained option to handle how those events are obtained and maybe even provide different events for different plugin managers. For now I'm fine if they are still coming from environment variables, just gathered in the `hosted-plugin.ts` during initialization of the manager and added as a dedicated, optional parameter.

What do you think?

Reply:
Sounds good, please have another look at the changes. 
I've moved the known events + reading the env var to `hosted-plugin.ts` and extended the `PluginManagerInitializeParams` to include the supported events.